### ISSUE TITLE
[BUG][Tabular] Unexpected crash in NN_FASTAI when fitting TabularPredictor on custom feature DataFrame

### ISSUE THREAD

**Issue by Killer3048** [autogluon/autogluon#5153]
When running the attached script to compute a set of normalized indicator features and then fit a `TabularPredictor` (with `presets='best_quality'` and `max_memory_usage_ratio=2`), the worker process dies unexpectedly with a “double free detected in tcache” / SIGABRT error. It is unclear why FastAI is aborting despite ample free RAM/VRAM.

**Observed Behavior**
```
free(): double free detected in tcache 2
*** SIGABRT received ***
Fatal Python error: Aborted
```

**LOGS:**
```
(_dystack pid=412296)   Fitting 8 child models (S1F1 - S1F8) | Fitting with ParallelLocalFoldFittingStrategy (8 workers, per: cpus=2, gpus=0, memory=1.22%)
(_dystack pid=412296) Dispatching folds on node c2e85730a7dda796e494b19456c0b15d7f68038c5949e067182ab0f3
(_ray_fit pid=438088) head node: True
(_ray_fit pid=438088) executing fold on node c2e85730a7dda796e494b19456c0b15d7f68038c5949e067182ab0f3
(_ray_fit pid=438088) ray worker training
(_ray_fit pid=438088)   Catboost model hyperparameters: {'iterations': 10000, 'learning_rate': 0.031375, 'random_seed': 0, 'allow_writing_files': False, 'eval_metric': 'Accuracy', 'depth': 8, 'grow_policy': 'Depthwise', 'l2_leaf_reg': 2.7998, 'max_ctr_complexity': 2, 'one_hot_max_size': 3, 'thread_count': 2}
(_ray_fit pid=438088) 0: learn: 0.6529851 test: 0.5625000 best: 0.5625000 (0) total: 55.4ms remaining: 9m14s
(_ray_fit pid=438088) 1: learn: 0.7022388 test: 0.5572917 best: 0.5625000 (0) total: 62.3ms remaining: 5m11s
(_ray_fit pid=438088) 2: learn: 0.7455224 test: 0.6041667 best: 0.6041667 (2) total: 74.1ms remaining: 4m06s

(raylet) A worker died or was killed while executing a task by an unexpected system error.  
RayTask ID: 83ead16e7e86707e214ad04797321f4f9f3c86e5 Worker ID: 10fdc76784cbb1a8d503bb7eddad04a7d19af4747f9eb2efa8dbb502 Node ID: c2e85730a7dda796e494b19456c0b15d7f68038c5949e067182ab0f3 Worker IP: 172.17.37.15 Worker port: 44273 Worker PID: 436758 Worker exit type: SYSTEM_ERROR Worker exit detail: Worker unexpectedly exits with a connection error code 2. End of file.

(raylet) A worker died or was killed while executing a task by an unexpected system error.  
RayTask ID: 168942cda521f3d1f64fad7c7f6cf34d0981fa91 Worker ID: 319f2b00b016ef202fa336a16b05ee3ae24a29ed83dc0289711827ae Node ID: c2e85730a7dda796e494b19456c0b15d7f68038c5949e067182ab0f3 Worker IP: 172.17.37.15 Worker port: 40811 Worker PID: 436941 Worker exit type: SYSTEM_ERROR Worker exit detail: Worker unexpectedly exits with a connection error code 2. End of file.

(_ray_fit pid=440329)   (11): Linear(in_features=213, out_features=2, bias=True) [repeated across cluster]
(_ray_fit pid=440329)   (8): Dropout(p=0.1189748, inplace=False) [repeated across cluster]
(_ray_fit pid=440329)   (softmax): Softmax(dim=1)
(_ray_fit pid=440329) Training tabular neural network for up to 1000 epochs...
(_ray_fit pid=440329) free(): double free detected in tcache 2
(_ray_fit pid=440329) *** SIGABRT received at time=1749120798 on cpu 15 ***
(_ray_fit pid=440329) PC: @     0x753e7c8969fc  (unknown)  pthread_kill
(_ray_fit pid=440329)     @     0x753e7c842520  (unknown)  (unknown)
(_ray_fit pid=440329) [2025-06-05 17:53:18,202 E 440329 440329] logging.cc:497: *** SIGABRT received at time=1749120798 on cpu 15 ***
(_ray_fit pid=440329) [2025-06-05 17:53:18,202 E 440329 440329] logging.cc:497: PC: @     0x753e7c8969fc  (unknown)  pthread_kill
(_ray_fit pid=440329) [2025-06-05 17:53:18,203 E 440329 440329] logging.cc:497:     @     0x753e7c842520  (unknown)  (unknown)
(_ray_fit pid=440329) Fatal Python error: Aborted
(_ray_fit pid=440329)   File ".../site-packages/torch/autograd/graph.py", line 823 in _engine_run_backward
(_ray_fit pid=440329)   File ".../site-packages/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py", line 387 in _train_net
(_ray_fit pid=440329)   File ".../site-packages/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py", line 245 in _fit
(_ray_fit pid=440329)   File ".../site-packages/autogluon/core/models/abstract/abstract_model.py", line 1051 in fit
(_ray_fit pid=440329)   File ".../site-packages/autogluon/core/models/ensemble/fold_fitting_strategy.py", line 413 in _ray_fit
(_ray_fit pid=440329)   File ".../site-packages/ray/_private/worker.py", line 945 in main_loop
(_ray_fit pid=440329)   File ".../site-packages/ray/_private/workers/default_worker.py", line 320 in <module>

```

**Installed Versions**
<details>
date                   : 2025-06-05
time                   : 17:50:49.305315
python                 : 3.10.15.final.0
OS                     : Linux
OS-release             : 6.6.87.1-microsoft-standard-WSL2
Version                : #1 SMP PREEMPT_DYNAMIC Mon Apr 21 17:08:54 UTC 2025
machine                : x86_64
processor              : x86_64
num_cores              : 16
cpu_ram_mb             : 40054.18359375
cuda version           : 12.575.57.04
num_gpus               : 1
gpu_ram_mb             : [8772]
avail_disk_size_mb     : 338788

accelerate             : 1.7.0
autogluon              : 1.3.1
autogluon.common       : 1.3.1
autogluon.core         : 1.3.1
autogluon.features     : 1.3.1
autogluon.multimodal   : 1.3.1
autogluon.tabular      : 1.3.1
autogluon.timeseries   : 1.3.1
boto3                  : 1.38.30
catboost               : 1.2.8
coreforecast           : 0.0.15
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.3
fastai                 : 2.8.2
fugue                  : 0.9.1
gluonts                : 0.16.1
huggingface-hub        : 0.32.4
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.6
joblib                 : 1.5.1
jsonschema             : 4.23.0
lightgbm               : 4.6.0
lightning              : 2.5.1.post0
matplotlib             : 3.10.3
mlforecast             : 0.13.6
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.8.1
numpy                  : 2.1.3
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.3.0
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.10.18
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.2.1
psutil                 : 7.0.0
pyarrow                : 20.0.0
pytesseract            : 0.3.13
pytorch-lightning      : 2.5.1.post0
pytorch-metric-learning: 2.8.1
ray                    : 2.44.1
requests               : 2.32.3
scikit-image           : 0.25.2
scikit-learn           : 1.6.1
scikit-learn-intelex   : None
scipy                  : 1.15.3
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.8.7
statsforecast          : 2.0.1
tabpfn                 : None
tensorboard            : 2.19.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.6.0
torchmetrics           : 1.7.2
torchvision            : 0.21.0
tqdm                   : 4.67.1
transformers           : 4.49.0
utilsforecast          : 0.2.10
xgboost                : 3.0.2
</details>

**Comment by Killer3048** [comment]
**Update**:   `num_gpus=1` and `num_cpus=16` in the `fit()` call resolves the “double free detected in tcache” error. For example:
```python
predictor.fit(
    train_df,
    presets='best_quality',
    verbosity=4,
    num_cpus=16,
    num_gpus=1, //**important for run without error**
    ag_args_fit={'max_memory_usage_ratio': 2}
)
```
With these settings, the Ray worker no longer crashes during the Tabular NN training
___
However, a new problem now appears during the weighted ensemble (level 3) stage:

When the ensemble reaches L3, the Ray worker never completes-instead, it appears to hang or crash and then immediately restarts the stacking process.
As a result, the fit procedure loops indefinitely through the level 3 weighted‐ensembling step, and no final model is produced.
In other words, L1 and L2 training finish successfully under these settings, but as soon as the code tries to build the weighted ensemble at L3, Ray fails to finalize the task and begins stacking again.

**Comment by Killer3048** [comment]
It turns out that the NVIDIA driver version 576.x on Windiws and matching WSL/Ubuntu driver was causing a `double free detected in tcache` crash inside `cudaGetDeviceCount` when running PyTorch models in parallel (via Ray/AutoGluon). Downgrading the Windows-side GPU driver to 572.83 (and ensuring WSL picked up the matching Linux driver) completely resolved the problem.

**Comment by Innixma** [comment]
Related: #5187
